### PR TITLE
Update OTel Go Client and Server examples

### PR DESCRIPTION
**Pull request by avillela**
- Updated OTel [`go/opentelemetry/client.go`](go/opentelemetry/client.go) and [`go/opentelemetry/server.go`](go/opentelemetry/server.go) examples to newer version of Go SDK.

Notes for @austinlparker:
- In [`go/opentelemetry/client.go`](go/opentelemetry/client.go), I create a span called `makeRequest` in the `makeRequest` function (see [here](https://github.com/lightstep/opentelemetry-examples/blob/f5722b17482acea1a22bfe845e4a8219316278ed/go/opentelemetry/client/client.go#L88-L93)), but that one span ends up disjointed from the overall trace. Can't figure out why. I don't think I'm initializing a new tracer, and I'm using the existing context, so whyyyyyy???
- I stuck with the original example, but cleaned it up a bit a, using [this](https://github.com/cartermp/opentelemetry-go/blob/cartermp.new-quickstart/website_docs/getting-started.md) as my source of code swipage and inspirationl.
- I added a [`collector/vanilla/config.yml`](collector/vanilla/config.yml) file to the `collector` directory. That way, folks following the docs can just use that config as the "quickie config" to send data to LS.

### CODE REVIEW

**Comment by codeboten** [comment]
@avillela i committed a change https://github.com/lightstep/opentelemetry-examples/pull/106/commits/69fb3e3acb04e6c71a596b688f854cca94e0c313 to start the `makeRequest` span before making the http call. PTAL to see if it results in what you'd expect to happen

File: docker-compose.yml [hunk 1]
```diff
@@ -43,24 +43,58 @@ services:
...
            - otel-collector
        env_file:
            - .env
        environment:
            - LS_SATELLITE_URL=otel-collector:4317
            - LS_INSECURE=1
-           - LS_SERVICE_NAME=go-opentelemetry-client
+           - LS_SERVICE_NAME=go-opentelemetry-client-collector
+           - LS_SERVICE_VERSION=9.9.2
+           - DESTINATION_URL=http://go-opentelemetry-server:8081/ping
```
Comment (codeboten) [inline docker-compose.yml:65]:
```suggestion
           - DESTINATION_URL=http://go-opentelemetry-server-collector:8081/ping
```

File: docker-compose.yml [hunk 2]
```diff
@@ -43,24 +43,58 @@ services:
...
+           - go-opentelemetry-server-otlp
+           - otel-collector
+       env_file:
+           - .env
+       environment:
+           - LS_SATELLITE_URL=otel-collector:4317
+           - LS_INSECURE=1
+           - LS_SERVICE_NAME=go-opentelemetry-client-otlp
            - LS_SERVICE_VERSION=9.9.2
            - DESTINATION_URL=http://go-opentelemetry-server:8081/ping
```
Comment (codeboten) [inline docker-compose.yml:99]:
```suggestion
           - DESTINATION_URL=http://go-opentelemetry-server-otlp:8081/ping
```

File: go/opentelemetry/otlp/server/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+COPY go.mod go.sum ./
+RUN go mod download
+COPY server.go .
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o server .
+
+FROM alpine:latest  
+RUN apk --no-cache add ca-certificates
+WORKDIR /root/
+COPY --from=0 /app/server .
+CMD ["./server"]
```
Comment (codeboten) [inline go/opentelemetry/otlp/server/Dockerfile:12]:
```suggestion
CMD ["./server"]

```

File: go/launcher/client/client-otlp.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
+//
+// example code to test lightstep/opentelemetry-exporter-go
```
Comment (codeboten) [inline go/launcher/client/client-otlp.go:2]: not sure what this comment block means. maybe it should be removed altogether

File: go/opentelemetry/collector/client/client.go [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+	if len(serviceVersion) == 0 {
+		serviceVersion = "0.1.0"
+		log.Printf("Using default service version %s", serviceVersion)
+	}
+
+	if len(lsEnvironment) == 0 {
+		lsEnvironment = "dev"
+		log.Printf("Using default environment %s", lsEnvironment)
+	}
```
Comment (codeboten) [inline go/opentelemetry/collector/client/client.go:75]: this variable isn't used anywhere

File: go/launcher/server/server-otlp.go [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+		log.Printf("Using default service name %s", serviceName)
+	}
+
+	if len(serviceVersion) == 0 {
+		serviceVersion = "0.1.0"
+		log.Printf("Using default service version %s", serviceVersion)
+	}
+
+	if len(lsToken) == 0 {
+		log.Fatalf("Lightstep token missing. Please set environment variable LS_ENVIRONMENT")
```
Comment (codeboten) [inline go/launcher/server/server-otlp.go:63]:
```suggestion
		log.Fatalf("Lightstep token missing. Please set environment variable LS_ACCESS_TOKEN")
```

Comment (avillela) [reply go/launcher/server/server-otlp.go]: Fixed

File: go/launcher/client/client-otlp.go (hunk 4 above)
Comment (avillela) [reply go/launcher/client/client-otlp.go:2]: Yeah, me neither. Was scratching my head on that one!

File: go/opentelemetry/collector/client/client.go (hunk 5 above)
Comment (avillela) [reply go/opentelemetry/collector/client/client.go:75]: Ooops...should've been referenced on line 92 and I instead hardcoded the value.

File: go/launcher/client/client-otlp.go (hunk 4 above)
Comment (avillela) [reply go/launcher/client/client-otlp.go]: Cleaned up the comment to make it more meaningful

File: docker-compose.yml (hunk 2 above)
Comment (avillela) [reply docker-compose.yml]: Fixed

File: docker-compose.yml (hunk 1 above)
Comment (avillela) [reply docker-compose.yml]: Fixed

**Review by codeboten — APPROVED** [review]

**Pull request merged**